test/topology_custom/test_mv_delete_partitions.py

```diff
@@ -0,0 +1,72 @@
+#
+# Copyright (C) 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+from test.pylib.manager_client import ManagerClient
+
+import asyncio
+import pytest
+import time
+import logging
+
+logger = logging.getLogger(__name__)
+
+async def insert_with_concurrency(cql, key, values, concurrency):
+    logger.info(f"Starting writes for key {key} with concurrency {concurrency}")
+    async def do_inserts(i: int):
+        # Insert all rows with keys with the same modulo as i
+        insert_stmt = cql.prepare(f"INSERT INTO ks.tab (key, c, v) VALUES (?, ?, ?)")
+        while i < values:
+            await cql.run_async(insert_stmt, [key, i, key + i])
+            # await cql.run_async(f"INSERT INTO ks.tab (key, c, v) VALUES ({key}, {i}, {key + i})")
+            i += concurrency
+    tasks = [asyncio.create_task(do_inserts(i)) for i in range(concurrency)]
+
+    await asyncio.gather(*tasks)
+    logger.info(f"Finished writes for key {key} with concurrency {concurrency}")
+
+async def wait_for_view(cql, ks, view, node_count):
+    while True:
+        done = 0
+        entries = await cql.run_async(f"SELECT host_id, status FROM system_distributed.view_build_status WHERE keyspace_name = '{ks}' AND view_name = '{view}'")
+        for entry in entries:
+            entry_list = list(entry)
+            if entry_list[1] == 'SUCCESS':
+                done += 1
+        if done == node_count:
+            return
+        else:
+            time.sleep(0.2)
+
+# This test reproduces issue #12379
+# The test scenario is the following:
+# 1. Create a table with 50 MVs
+# 2. Insert 50000 rows into 50 partitions (1000 rows per partition)
+# 3. Wait for all MVs to be built
+# 4. Delete all partitions, causing 50*40*1000 = 2M view updates
+# If the test doesn't cause a crash due to OOM in the step 4, it passes
+@pytest.mark.asyncio
+async def test_delete_partition_rows_from_table_with_mv(manager: ManagerClient) -> None:
+    await manager.servers_add(2, cmdline=['-m', '400MB'])
```

Comment:
> How long does this test take? It seems like it can take a very long time, and potentially a VERY VERY long time in debug mode (did you check?), perhaps to the point it should be skipped in debug mode.

It takes about a minute on my machine in dev and release modes and I'll actually skip this test in debug mode simply because the `-m` option doesn't actually work there and the test would in fact take way too long if it used the default amount of memory.

> It's nice that you can run Scylla with decreased memory. So, I wonder, can you decrease it even more? 100MB? Even less? Maybe this will allow you to reproduce this bug with even fewer views and rows.
> Also, did you check that these numbers (50 views, gazillion rows) are the (roughly) the minimum that reproduces this issue? If using half also reproduces it, it's better to use half.

I managed to lower the memory to 300MB which leaves 100MB per shard, I couldn't reach the partition deletes in the test with anything less.
It might be possible to get the numbers slightly lower but the test already isn't that long anymore and the results might be less consistent with anything less so I'd say we leave it at that

Reply:
It's sad not to run in debug mode, it's good at finding bugs.

Can you reduce the thresholds instead so throttling triggers earlier?